To: vim_dev@googlegroups.com
Subject: Patch 7.3.154
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.154 (after 7.3.148)
Problem:    Can't compile with tiny features. (Tony Mechelynck)
Solution:   Move #define outside of #ifdef.
Files:	    src/syntax.c


*** ../vim-7.3.153/src/syntax.c	2011-04-01 14:44:54.000000000 +0200
--- src/syntax.c	2011-04-02 15:11:47.000000000 +0200
***************
*** 68,73 ****
--- 68,75 ----
  
  #define HL_TABLE() ((struct hl_group *)((highlight_ga.ga_data)))
  
+ #define MAX_HL_ID       20000	/* maximum value for a highlight ID. */
+ 
  #ifdef FEAT_CMDL_COMPL
  /* Flags to indicate an additional string for highlight name completion. */
  static int include_none = 0;	/* when 1 include "None" */
***************
*** 225,236 ****
   * 22000 - 22999  CONTAINED indicator (current_syn_inc_tag added)
   * 23000 - 32767  cluster IDs (subtract SYNID_CLUSTER for the cluster ID)
   */
! #define SYNID_ALLBUT	20000	    /* syntax group ID for contains=ALLBUT */
  #define SYNID_TOP	21000	    /* syntax group ID for contains=TOP */
  #define SYNID_CONTAINED	22000	    /* syntax group ID for contains=CONTAINED */
  #define SYNID_CLUSTER	23000	    /* first syntax group ID for clusters */
  
- #define MAX_SYNID       SYNID_ALLBUT
  #define MAX_SYN_INC_TAG	999	    /* maximum before the above overflow */
  #define MAX_CLUSTER_ID  (32767 - SYNID_CLUSTER)
  
--- 227,237 ----
   * 22000 - 22999  CONTAINED indicator (current_syn_inc_tag added)
   * 23000 - 32767  cluster IDs (subtract SYNID_CLUSTER for the cluster ID)
   */
! #define SYNID_ALLBUT	MAX_HL_ID   /* syntax group ID for contains=ALLBUT */
  #define SYNID_TOP	21000	    /* syntax group ID for contains=TOP */
  #define SYNID_CONTAINED	22000	    /* syntax group ID for contains=CONTAINED */
  #define SYNID_CLUSTER	23000	    /* first syntax group ID for clusters */
  
  #define MAX_SYN_INC_TAG	999	    /* maximum before the above overflow */
  #define MAX_CLUSTER_ID  (32767 - SYNID_CLUSTER)
  
***************
*** 6462,6468 ****
  
  #endif /* FEAT_SYN_HL */
  
- 
  /**************************************
   *  Highlighting stuff		      *
   **************************************/
--- 6463,6468 ----
***************
*** 8996,9004 ****
  	highlight_ga.ga_growsize = 10;
      }
  
!     if (highlight_ga.ga_len >= MAX_SYNID)
      {
! 	EMSG(_("E849: Too many syntax groups"));
  	vim_free(name);
  	return 0;
      }
--- 8996,9004 ----
  	highlight_ga.ga_growsize = 10;
      }
  
!     if (highlight_ga.ga_len >= MAX_HL_ID)
      {
! 	EMSG(_("E849: Too many highlight and syntax groups"));
  	vim_free(name);
  	return 0;
      }
*** ../vim-7.3.153/src/version.c	2011-04-02 14:44:50.000000000 +0200
--- src/version.c	2011-04-02 14:52:33.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     154,
  /**/

-- 
BEDEVERE:        Why do you think she is a witch?
SECOND VILLAGER: She turned me into a newt.
BEDEVERE:        A newt?
SECOND VILLAGER: (After looking at himself for some time) I got better.
                 "Monty Python and the Holy Grail" PYTHON (MONTY) PICTURES LTD

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
